### PR TITLE
[FIX] db-update issues when switching to new release_* branches

### PR DESCRIPTION
**Pull request by chfsx**
This PR solves several issues concerning the database when switching from old versions (prior to 5.6) to release_7 (the same PR will be introduced for release_8 as well). See #251 https://github.com/opencast-ilias/OpenCast/issues/252#issuecomment-1975867210

@ferishili thanks for your help and @reiferschris for the finding as well!

### CODE REVIEW

File: sql/dbupdate.php [hunk 1]
```diff
@@ -1583,3 +1610,95 @@
...
+
+if(!$ilDB->tableColumnExists('xoct_workflow', 'config_panel')) {
+    $ilDB->addTableColumn('xoct_workflow', 'config_panel', [
+        'type' => 'clob',
+    ]);
+}
+
+?>
+<?php
+// Dear colleague!
```
Comment (chfsx) [inline sql/dbupdate.php:1677]: @ferishili I introduced this comment here to remind ourself and other to discuss the introduction of new steps :-)

Comment (ferishili) [reply sql/dbupdate.php:1677]: Well explained, thank you very much.

File: sql/dbupdate.php [hunk 2]
```diff
@@ -1555,14 +1555,41 @@
 ?>
 <#44>
 <?php
-// To apply new changes into WorkflowAP model as well as xoct_workflow table.
-//\srag\Plugins\Opencast\Model\Workflow\WorkflowAR::updateDB();
+// Re remove the column parameter from xoct_workflow if it's existent.
+if($ilDB->tableColumnExists('xoct_workflow', 'parameter')) {
```
Comment (ferishili) [inline sql/dbupdate.php]: The field is called `parameters` and the fix has `parameter`
This also applies to the latest step!

**Review by ferishili — CHANGES_REQUESTED** [review]

File: sql/dbupdate.php [hunk 3]
```diff
@@ -1583,3 +1610,95 @@
...
+        [$new, $old]
+    );
+}
+?>
+<#48>
+<?php
+// Since there can be cases where step 44 was not executed the right way, we must ensure that the changes to the xoct_workflow table are applied.
+
+// Re remove the column parameter from xoct_workflow if it's existent.
+if($ilDB->tableColumnExists('xoct_workflow', 'parameter')) {
```
Comment (ferishili) [inline sql/dbupdate.php]: Here as well: `parameters`

**Review by ferishili — CHANGES_REQUESTED** [review]

**Review by ferishili — APPROVED** [review]

**Comment by ferishili** [comment]
Well done, thanks a lot

**Comment by dagraf** [comment]
I close this PR, since it got included in #258 and #255.

**Pull request closed without merging**